include/cglm/types-struct.h

```diff
@@ -33,4 +33,62 @@ typedef union CGLM_ALIGN_IF(16) vec4s {
   vec4 raw;
 } vec4s;
 
+typedef vec4s versors;
+
+typedef union mat3s {
+#ifndef CGLM_NO_ANONYMOUS_STRUCT
+  struct {
+    float m00;
+    float m01;
+    float m02;
+    float m10;
+    float m11;
+    float m12;
+    float m20;
+    float m21;
+    float m22;
+  };
```

Comment:
Don't forget that **cglm** stores matrices as column-major order. So second item must be `m10` not `m01` right? Am I wrong? 

Also formatting that like below could help to keep order correct and more readable:

```C
struct {
  float m00, m10, m20,
        m01, m11, m21,
        m02, m12, m22;
};

/* or */

struct {
  float m00, m10, m20; /* first column  */
  float m01, m11, m21; /* second column */
  float m02, m12, m22; /* third column  */
};
```

What does **m01** mean? First column's first item or first row's first item? What did you think? Or I must do some research for this :S

Reply:
maybe `m.m01` should be identical to `m[0][1]`. So it could be:

```C
struct {
  float m00, m01, m02; /* first column  */
  float m10, m11, m12; /* second column */
  float m20, m21, m22; /* third column  */
};
```

in this case, your implementation seems ok.

Reply:
Well, what I thought about this is that in reality it doesn't matter as long the consistency is maintained. But your last comment make it more clear, it's true that `m.m01` should be the same as `m.raw[0][1]`.

So, should I keep it as:
```c
struct {
  float m00, m01, m02,
        m10, m11, m12,
        m20, m21, m22;
};
```

Reply:
More thinking about this:

Really the dominant field in the `mat3s` is the `mat3 raw;` field, those are defined as:
```c
vec3 mat3[3];
```
so, when the user do: `m[0][1]` is really accessing the second element of the first column (the second element of the first `vec3`), so every other field in the `mat3s` struct/union should reflect this:
```c
struct {
  float m00, m10, m20,
        m01, m11, m21,
        m02, m21, m22
};
```
All of this matter on how you treat matrices in the code, specially in the multiplication of a matrix and vector, and since you treat them as column ordered, it should be like you put it at the start of this thread.

Reply:
Did you mean this:

```C
struct {
  vec3s col1,  col2, col3; /* you used vec4s here */
}
```

in this case since there is no alignment on vec3s, it should match, am I wrong? I'll not define rows like that since we only have columns. But something like this could be possible:

```C
typedef union mat3s {
#ifndef CGLM_NO_ANONYMOUS_STRUCT
  /* row accessor */
  struct row {
    float m00, m10, m20; /* first row  */
    float m01, m11, m21; /* second row */
    float m02, m12, m22; /* third row  */
  };

  /* column accessor (DEFAULT) */
  struct {
    float m00, m01, m02; /* first column  */
    float m10, m11, m12; /* second column */
    float m20, m21, m22; /* third column  */
  };
  
  /* column vector accessor */
  struct {
    vec3s col0;
    vec3s col1;
    vec3s col2;
  };
#endif
  mat3 raw;
} mat3s;

```

Example:

```C
m3.m01.    /* first column's first item: same as m3.raw[0][1] */
m3.row.m01 /* first row's first item */
m3.col1    /* first column */
```

I don't say that we should that, just thoughts...

Reply:
Yeah, that makes sense to me. Don't know about providing a row fields just now since the entire API is based on column-major matrices. Maybe if it's useful to some people in a real case scenario that struct can be included later. Don't know, your call.

Reply:
> Maybe if it's useful to some people in a real case scenario that struct can be included later. Don't know, your call.

Let's postpone to add .row selector/accessor.

For this struct:

```C
struct {
    vec3s col0;
    vec3s col1;
    vec3s col2;
};
```

which one is better:

```C
m.col0
m.col1
m.col2

/* vec3s col[3]; instead of anonymous struct */
m.col[0]
m.col[1]
m.col[2]
```

what do you think?

Reply:
hmm... both?:
```c
typedef union mat3s {
#ifndef CGLM_NO_ANONYMOUS_STRUCT
  /* column accessor */
  struct {
    float m00, m01, m02; /* first column  */
    float m10, m11, m12; /* second column */
    float m20, m21, m22; /* third column  */
  };
  
  /* column vector accessor */
  struct {
    vec3s col0;
    vec3s col1;
    vec3s col2;
  };
#endif

  /* column array accessor */
  vec3s col[3];

  /* raw accessor */
  mat3 raw;
} mat3s;
```

Reply:
No, not both.

Reply:
Well, look a the `glms_frustum_planes` implementation, how do you like it better?
```c
CGLM_INLINE
void
glms_frustum_planes(mat4s m, vec4s dest[6]) {
  mat4s t;

  t = glms_mat4_transpose(m);

  dest[0] = glms_vec4_add(t.col[3], t.col[0]); /* left   */
  dest[1] = glms_vec4_sub(t.col[3], t.col[0]); /* right  */
  dest[2] = glms_vec4_add(t.col[3], t.col[1]); /* bottom */
  dest[3] = glms_vec4_sub(t.col[3], t.col[1]); /* top    */
  dest[4] = glms_vec4_add(t.col[3], t.col[2]); /* near   */
  dest[5] = glms_vec4_sub(t.col[3], t.col[2]); /* far    */

  ...
}
```
or
```c
CGLM_INLINE
void
glms_frustum_planes(mat4s m, vec4s dest[6]) {
  mat4s t;

  t = glms_mat4_transpose(m);

  dest[0] = glms_vec4_add(t.col[3], t.col0); /* left   */
  dest[1] = glms_vec4_sub(t.col[3], t.col0); /* right  */
  dest[2] = glms_vec4_add(t.col[3], t.col1); /* bottom */
  dest[3] = glms_vec4_sub(t.col[3], t.col1); /* top    */
  dest[4] = glms_vec4_add(t.col[3], t.col2); /* near   */
  dest[5] = glms_vec4_sub(t.col[3], t.col2); /* far    */

  ...
}
```

if must be one of another, I think we should go with `t.col[0]` because it offer the possibility of `t.col[i]` where `i` is a variable previously calculated.

Reply:
Then let's go with array (`col[i]`) for column accessors